src/search/cost_test_ebpf.cc

```diff
@@ -13,7 +13,14 @@ double get_error_cost(inst* p1, inst* p2, int win_start, int win_end) {
 
   inout_t::start_insn = win_start;
   inout_t::end_insn = win_end;
-  static_safety_check_pgm(prog1.inst_list, inst::max_prog_len);
+  int num_static_unsafe_ins = static_safety_check_pgm(prog1.inst_list, inst::max_prog_len, true);
+  if (num_static_unsafe_ins > 0){
+    // stop execution here since the p1 is unsafe
+    // error message would have already been thrown by static_safety_check_pgm
+    // so throw empty err message to stop execution
+    string err_msg = "";
+    throw(err_msg);
+  }
```

Comment:
I didn't follow why this function _needs_ to throw an error message... could you explain?

Reply:
In the original k2 compiler code the function static_safety_check_pgm had a void return value. The function either threw an error (when an unsafe instruction was identified) or executed without errors (no unsafe instructions observed during static analysis).

However, now the functionality has changed to handle the error within the static_safety_check_pgm and return an integer value.

Over here, the original k2 program had a single line of code calling the static_safety_chech_pgm function. Which means that either the program passed the check and moved on or this function stops the whole program immediately by throwing an error.

Now, in order to maintain this behavior I had to replace the single static_safety_check_pgm call with these lines you have commented on. 

If the program is unsafe, the error is handled and printed within the call but the code here should still stop like before. So, I threw an empty error message.

Same reasoning here applies to the your other feedbacks of this issue.

I think maybe this static_safety_check function might change as we try other ways to granulize safety cost so based on how things end up later we could make this code more clean accordingly. But do let me know if there are any suggestions to make it clean now itself. I'll make the changes!

Reply:
Thanks for that clarification. Perhaps the error messages could be made more self-explanatory (rather than empty). Otherwise, i'm okay with this design.